doc/src/contributing/new-contributors-guide/writing-tests.md

```diff
@@ -0,0 +1,1078 @@
+
+Consequently, all functionality in SymPy is tested extensively. This guide
+goes over how tests in SymPy are written.
+
+## Testing Policies
+
+In order to ensure the high standard of correctness, SymPy has the following
+rules that apply to all pull requests:
+
+1. All new functionality must be tested. Tests should aim to cover all
+   possible cases to best ensure correctness. This means not only maximizing
+   code coverage, but also covering all possible corner cases.
+
+2. Every pull request must pass all tests before it can be merged. The tests
+   are automatically run by the GitHub Actions CI on every pull request. If
+   any tests fail, the CI will fail with a red ❌. These failures must be
+   addressed before the pull request can be merged.
+
+3. Bug fixes should be accompanied by a [regression
+   test](writing-tests-regression-tests).
+
+(writing-tests-basics)=
+## Basics for Writing Tests
+
+Tests are located alongside the code in `tests/` directories, in files named
+`test_<thing>.py`. In most cases, if you modified
+`sympy/<submodule>/<file>.py` then the test for the functionality will go in
+`sympy/<submodule>/tests/test_<file>.py`. For example, the tests for the
+functions in `sympy/simplify/sqrtdenest.py` are in
+`sympy/simplify/tests/test_sqrtdenest.py`. There are some exceptions to this
+rule, so in general try to find where the existing tests are for a function
+and add your tests alongside them. If you are adding tests for a new function,
+follow the general pattern of tests in the module you are adding to.
+
+Tests follow a simple pattern, which should be apparent from reading the
+existing test files. Tests are in functions that start with `test_` and
+contain lines like
+
+```py
+assert function(arguments) == result
+```
+
+For example
+
+```py
+# from sympy/functions/elementary/tests/test_trigonometric.py
+
+def test_cos_series():
+    assert cos(x).series(x, 0, 9) == \
+        1 - x**2/2 + x**4/24 - x**6/720 + x**8/40320 + O(x**9)
+```
+
+New test cases can be added to an existing test function if it is relevant, or
+you can create a new test function.
+
+## Running Tests
+
+The basic way to run the tests is to use
+
+```
+./bin/test
+```
+
+to run the tests, and
+
+```
+./bin/doctest
+```
+
+to run the doctests. Note that the full test suite can take some time to run,
+so you may prefer to just run a subset of the tests, e.g., corresponding to
+the module you modified. You can do this by passing the name of the submodules
+or tests files to the test command. For example,
+
+```
+./bin/test solvers
+```
+
+will run only the tests for the solvers.
+
+If you want, you can also use `pytest` to run the tests instead of the
+`./bin/test` tool.
+
+Another option is to just push your code up to GitHub and let the tests run on
+the CI. The GitHub Actions CI will run all the tests. However, it can take
+some time to finish, so it is usually advisable to run at least the basic
+tests before committing to avoid having to wait.
```

Comment:
Perhaps an example here of how to interpret the output from CI and then run the test locally e.g. if you see this:
```
_____________________________________________________________________________________________________
_________________ sympy/printing/pretty/tests/test_pretty.py:test_upretty_sub_super _________________
Traceback (most recent call last):
  File "/home/oscar/current/sympy/sympy.git/sympy/printing/pretty/tests/test_pretty.py", line 317, in test_upretty_sub_super
    assert upretty( Symbol('beta_1_2') ) == 'β₁₂'
AssertionError
```
then you can reproduce the failure locally with (note the extra colon):
```
$ pytest sympy/printing/pretty/tests/test_pretty.py::test_upretty_sub_super
...
______________________________________ test_upretty_sub_super _______________________________________

    def test_upretty_sub_super():
>       assert upretty( Symbol('beta_1_2') ) == 'β₁₂'
E       AssertionError: assert 'β₁ ₂' == 'β₁₂'
E         - β₁₂
E         + β₁ ₂
E         ?   +

sympy/printing/pretty/tests/test_pretty.py:317: AssertionError
```

Reply:
It's a simple fix to make the SymPy test runner both accept and produce pytest-style double colon test specifications, so I've made that change here.